processing/src/main/java/org/apache/carbondata/processing/loading/steps/CarbonRowDataWriterProcessorStepImpl.java

```diff
@@ -128,16 +128,18 @@ public CarbonRowDataWriterProcessorStepImpl(CarbonDataLoadConfiguration configur
       CarbonTimeStatisticsFactory.getLoadStatisticsInstance()
           .recordDictionaryValue2MdkAdd2FileTime(CarbonTablePath.DEPRECATED_PARTITION_ID,
               System.currentTimeMillis());
-
+      ExecutorService fallBackExecutorService =
+          Executors.newFixedThreadPool(1, new CarbonThreadFactory("FallBackPool:"));
```

Comment:
pool size cannot be always one ...please refer 
org/apache/carbondata/processing/store/writer/AbstractFactDataWriter.java:203